### DISCUSSION TITLE
What is the exact definition of many-to-many, any-to-many etc in voice conversion task?

### DISCUSSION THREAD

**Discussion by alifarrokh** [coqui-ai/TTS#1784] [General Q&A] [answered]
Hi.

I've read some of the papers related to the voice conversion task, including the YourTTS paper, and I'm a little bit confused about the meaning of _many-to-many model_, _any-to-many model_ etc. Can anyone please give me an exact definition of these expressions?

- one-to-many
- many-to-many
- any-to-many
- any-to-any
- etc.

Thanks in advance.

**Comment by p0p4k** [comment]
Can you point out the article/paper (line number and link) where these terms come up? If it is about seq2seq models, the definition is gonna be straightforward.

**Reply by alifarrokh** [reply]
Here are a few examples:
- StarGAN-VC: Non-parallel ***many-to-many*** voice conversion with star generative adversarial networks [[pdf]](https://arxiv.org/abs/1806.02169)
- Cotatron: Transcription-Guided Speech Encoder for ***Any-to-Many*** Voice Conversion without Parallel Data [[pdf]](https://arxiv.org/abs/2005.03295)
- FragmentVC: ***Any-to-Any*** Voice Conversion by End-to-End Extracting and Fusing Fine-Grained Voice Fragments With Attention [[pdf]](https://arxiv.org/abs/2010.14150)

**Comment by alifarrokh** [comment, marked as answer]
This is what I found after comparing some papers and their demo audios.
The `X` in `X-to-Y` refers to the input audio and the `Y` refers to the output audio. Also:
- `one`: the input/output audio should/will be from a specific speaker that the models is trained with.
- `many`: the input/output audio can be from a list of known speakers the model has seen during the training.
- `any`: the input/output audio can be from any arbitrary speaker.

**Reply by p0p4k** [reply]
Good to know. Thanks for sharing.
